src/ngResource/resource.js

```diff
@@ -721,7 +718,13 @@ angular.module('ngResource', ['ng']).
                 (success || noop)(value, response.headers);
                 return value;
               },
-              responseErrorInterceptor);
+              responseErrorInterceptor || error ?
+              function(response) {
+                (error || noop)(response);
+                (responseErrorInterceptor || noop)(response);
+                return response;
```

Comment:
The alternative would be to change `return $q.reject(response);` to `return responseErrorInterceptor ? $q.reject(response) : response;` that I think it is a little bit too much